lib/storage/storage.go

```diff
@@ -324,41 +324,48 @@ func (s *Storage) CreateSnapshot() (string, error) {
 		return "", fmt.Errorf("cannot create dir %q: %w", dstDataDir, err)
 	}
 
-	smallDir, bigDir, err := s.tb.CreateSnapshot(snapshotName)
+	smallDir, bigDir, err := s.tb.CreateSnapshot(snapshotName, deadline)
 	if err != nil {
+		fs.MustRemoveAll(dstDir)
```

Comment:
nit: what about wrap the func body into another func and check for returned error only once?